akka-cluster/src/test/scala/akka/cluster/DowningProviderSpec.scala

```diff
@@ -0,0 +1,92 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.cluster
+
+import java.util.concurrent.atomic.AtomicBoolean
+
+import akka.ConfigurationException
+import akka.actor.{ ActorSystem, Props }
+import akka.testkit.TestKit.{ awaitCond, shutdownActorSystem }
+import akka.testkit.{ TestKit, TestProbe }
+import com.typesafe.config.ConfigFactory
+import org.scalatest.concurrent.Futures
+import org.scalatest.{ Matchers, WordSpec }
+
+import scala.concurrent.duration._
+
+class FailingDowningProvider(system: ActorSystem) extends DowningProvider {
+  override val downRemovalMargin: FiniteDuration = 20.seconds
+  override def downingActorProps: Option[Props] = {
+    throw new ConfigurationException("this provider never works")
+  }
+}
+
+class DummyDowningProvider(system: ActorSystem) extends DowningProvider {
+  override val downRemovalMargin: FiniteDuration = 20.seconds
+
+  val actorPropsAccessed = new AtomicBoolean(false)
+  override val downingActorProps: Option[Props] = {
+    actorPropsAccessed.set(true)
+    None
+  }
+}
+
+class DowningProviderSpec extends WordSpec with Matchers {
+
+  val baseConf = ConfigFactory.parseString(
+    """
+      akka {
+        loglevel = WARNING
+        actor.provider = "akka.cluster.ClusterActorRefProvider"
+        remote {
+          netty.tcp {
+            hostname = "127.0.0.1"
+            port = 0
+          }
+        }
+      }
+    """).withFallback(ConfigFactory.load())
+
+  "The downing provider mechanism" should {
+
+    "default to akka.cluster.NoDowning" in {
+      val system = ActorSystem("default", baseConf)
+      Cluster(system).downingProvider shouldBe an[NoDowning]
+      shutdownActorSystem(system)
+    }
+
+    "use akka.cluster.AutoDowning if 'auto-down-unreachable-after' is configured" in {
+      val system = ActorSystem("auto-downing", ConfigFactory.parseString(
+        """
+          akka.cluster.auto-down-unreachable-after = 18d
```

Comment:
YES, finally someone that understands that test timeouts should not be too short! :-)